wal-routing/src/router/mod.rs

```diff
@@ -0,0 +1,236 @@
+pub mod builder;
+
+use std::{cell::RefCell, collections::HashMap, rc::Rc};
+
+use gloo::utils::{body, history, window};
+use wal::{component::node::AnyComponentNode, virtual_dom::dom};
+use wasm_bindgen::{prelude::Closure, JsCast, JsValue};
+use web_sys::{Element, Event, EventTarget};
+
+pub struct PageRenderer {
+    generator: Box<dyn Fn() -> Rc<RefCell<AnyComponentNode>>>,
+}
+
+impl PageRenderer {
+    pub fn new(generator: impl Fn() -> Rc<RefCell<AnyComponentNode>> + 'static) -> PageRenderer {
+        PageRenderer {
+            generator: Box::new(generator),
+        }
+    }
+
+    pub fn render(&self) -> Rc<RefCell<AnyComponentNode>> {
+        (*self.generator)()
+    }
+}
+
+thread_local! {
+    pub static ROUTER: RefCell<Router> = RefCell::new(Router::mock());
+}
+
+const WAL_ROUTING_ATTR: &'static str = "data_link";
+
+pub struct Router {
+    pages: HashMap<&'static str, PageRenderer>,
+    error_path: &'static str,
+    cur_path: String,
+    cur_page: Option<Rc<RefCell<AnyComponentNode>>>,
+}
+
+impl Router {
+    pub(crate) fn mock() -> Router {
+        Router {
+            pages: [].into(),
+            error_path: "undefined",
```

Comment:
undefined == None? Czy to ma inne znaczenie bo jeśli nie to może error_path i cur_path to powinny być optiony? A jak to znaczy co innego to można wydzielić przynajmniej do stałej